desktop/libs/liboozie/src/liboozie/submission2.py

```diff
@@ -616,10 +616,16 @@ def _is_coordinator(self):
 
   def _create_file(self, deployment_dir, file_name, data, do_as=False):
     file_path = self.fs.join(deployment_dir, file_name)
+
+    # In Py3 because of i18n, the xml data is not properly utf-8 encoded for some languages.
+    # This can later throw UnicodeEncodeError exception for request body in HDFS or other FS API calls. To tackle this,
+    # We are converting the data into bytes by utf-8 encoding instead of str type.
+    data = smart_str(data).encode('utf-8') if sys.version_info[0] > 2 else smart_str(data)
```

Comment:
Does it break if we use `encode` even for py2? That would help keep the 2vs3 logic simpler.

Reply:
Yes, if we use `.encode('utf-8')` in py2 Hue, we ae getting the following error:
```
Workflow 제출 실패: 'ascii' codec can't decode byte 0xec in position 134: ordinal not in range(128)
```

Regarding 2vs3 logic, once all form factors are supporting py3, we can slowly remove py2 code support in Hue. This is also one of our roadmap items @amitsrivastava.